From 63edea5b7e5676698d66cb88af6233309abe4c91 Mon Sep 17 00:00:00 2001
From: Marius Brehler <marius.brehler@amd.com>
Date: Tue, 15 Jul 2025 09:48:47 +0000
Subject: [PATCH 2/2] Find bundled libelf

---
 cmake/rocprofiler_config_interfaces.cmake | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake/rocprofiler_config_interfaces.cmake b/cmake/rocprofiler_config_interfaces.cmake
index b41780f6..6a548648 100644
--- a/cmake/rocprofiler_config_interfaces.cmake
+++ b/cmake/rocprofiler_config_interfaces.cmake
@@ -204,8 +204,8 @@ target_link_libraries(rocprofiler-sdk-ptl INTERFACE PTL::ptl-static)
 #
 # ----------------------------------------------------------------------------------------#
 
-find_package(libelf REQUIRED)
-target_link_libraries(rocprofiler-sdk-elf INTERFACE libelf::libelf)
+find_package(LibElf REQUIRED)
+target_link_libraries(rocprofiler-sdk-elf INTERFACE elf::elf)
 
 # ----------------------------------------------------------------------------------------#
 #
-- 
2.43.0

