From 81af9fa868610b043d0a4093433f98fc2c7c2ac5 Mon Sep 17 00:00:00 2001
From: Stella Laurenzo <stellaraccident@gmail.com>
Date: Mon, 14 Apr 2025 15:24:10 -0700
Subject: [PATCH 4/4] Replace deprecated GTest::GTest with GTest::gtest.

---
 clients/gtest/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/clients/gtest/CMakeLists.txt b/clients/gtest/CMakeLists.txt
index a3566f9..cdbf9be 100644
--- a/clients/gtest/CMakeLists.txt
+++ b/clients/gtest/CMakeLists.txt
@@ -110,7 +110,7 @@ target_link_libraries(hipsolver-test PRIVATE
   ${LAPACK_LIBRARIES}
   hipsolver-common
   clients-common
-  GTest::GTest
+  GTest::gtest
   Threads::Threads
   roc::hipsolver
 )
-- 
2.43.0

