From fcdd342f60ecd6a4b5544f999819071f7c532e37 Mon Sep 17 00:00:00 2001
From: Scott <scott.todd0@gmail.com>
Date: Thu, 1 May 2025 10:59:02 -0700
Subject: [PATCH 5/5] Check if rocblas_DIR env var is set before using it.

---
 clients/gtest/CMakeLists.txt | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/clients/gtest/CMakeLists.txt b/clients/gtest/CMakeLists.txt
index cdbf9be..41d003b 100644
--- a/clients/gtest/CMakeLists.txt
+++ b/clients/gtest/CMakeLists.txt
@@ -172,9 +172,13 @@ if(WIN32)
       ARGS -E copy ${file_i} ${PROJECT_BINARY_DIR}/staging/
     )
   endforeach()
-  add_custom_command(TARGET hipsolver-test
-    POST_BUILD
-    COMMAND ${CMAKE_COMMAND}
-    ARGS -E copy_directory $ENV{rocblas_DIR}/bin/rocblas/library ${PROJECT_BINARY_DIR}/staging/library
-  )
+  if(DEFINED $ENV{rocblas_DIR})
+    add_custom_command(TARGET hipsolver-test
+      POST_BUILD
+      COMMAND ${CMAKE_COMMAND}
+      ARGS -E copy_directory $ENV{rocblas_DIR}/bin/rocblas/library ${PROJECT_BINARY_DIR}/staging/library
+    )
+  else()
+    message(WARNING "ENV{rocblas_DIR} not set, tests may be missing .dlls")
+  endif()
 endif()
-- 
2.47.1.windows.2

