From 206959ef7d064c5f4a2a8afc11979fc942180347 Mon Sep 17 00:00:00 2001
From: Stella Laurenzo <stellaraccident@gmail.com>
Date: Fri, 2 May 2025 15:27:13 -0700
Subject: [PATCH 3/3] Limit number of Tensile workers to 61 on Windows due to
 system limit.

---
 Tensile/Parallel.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/Tensile/Parallel.py b/Tensile/Parallel.py
index a643c1ae..f483d264 100644
--- a/Tensile/Parallel.py
+++ b/Tensile/Parallel.py
@@ -36,7 +36,10 @@ def CPUThreadCount(enable=True):
         return 1
     else:
         if os.name == "nt":
-            cpu_count = os.cpu_count()
+            # Windows supports at most 61 workers because the scheduler uses
+            # WaitForMultipleObjects directly, which has the limit (the limit
+            # is actually 64, but some handles are needed for accounting).
+            cpu_count = min(os.cpu_count(), 61)
         else:
             cpu_count = len(os.sched_getaffinity(0))
         cpuThreads = globalParameters["CpuThreads"]
-- 
2.41.0.windows.1

