From f90b28f1d1a25a1a92d82d403e42d9fb3e5d27ab Mon Sep 17 00:00:00 2001
From: Scott Todd <scott.todd0@gmail.com>
Date: Thu, 5 Jun 2025 12:12:40 -0700
Subject: [PATCH 3/3] Re-apply Windows compatibility fixes to command
 construction.

---
 Tensile/cmake/TensileConfig.cmake | 22 ++++++++++++++++++----
 1 file changed, 18 insertions(+), 4 deletions(-)

diff --git a/Tensile/cmake/TensileConfig.cmake b/Tensile/cmake/TensileConfig.cmake
index 7fe89bf4..f905935f 100644
--- a/Tensile/cmake/TensileConfig.cmake
+++ b/Tensile/cmake/TensileConfig.cmake
@@ -215,6 +215,9 @@ function(TensileCreateLibraryFiles
     set(Options ${Options} "--architecture=${archString}")
   endif()
 
+  # We do not need to do device enumeration at library build time.
+  set(Options ${Options} "--no-enumerate")
+
   set(CommandLine ${Script} ${Options} ${Tensile_LOGIC_PATH} ${Tensile_OUTPUT_PATH} HIP)
   if (WIN32 OR (VIRTUALENV_BIN_DIR AND VIRTUALENV_PYTHON_EXENAME))
     set(CommandLine ${VIRTUALENV_BIN_DIR}/${VIRTUALENV_PYTHON_EXENAME} ${CommandLine})
@@ -222,8 +225,13 @@ function(TensileCreateLibraryFiles
   # Tensile relies on the tools from the path, so capture the configure time
   # path. It would be better if this were explicit, but that would be a pretty
   # big change.
+  set(ESC_PATH "$ENV{PATH}")
+  if(WIN32)
+    string(REPLACE ";" "$<SEMICOLON>" ESC_PATH "${ESC_PATH}")
+  endif()
+  set(ENV_PATH_ARG "PATH=${ESC_PATH}")
   set(CommandLine
-    "${CMAKE_COMMAND}" -E env "'PATH=$ENV{PATH}'" --
+    "${CMAKE_COMMAND}" -E env "PATH=${ESC_PATH}" --
     ${CommandLine})
   message(STATUS "Tensile_CREATE_COMMAND: ${CommandLine}")
 
@@ -258,12 +266,17 @@ function(TensileCreateLibraryFiles
         OUTPUT "${Tensile_OUTPUT_PATH}/library"
         DEPENDS ${Tensile_LOGIC_PATH}
         COMMAND ${CommandLine}
-        COMMENT "Generating libraries with TensileCreateLibrary")
+        COMMENT "Generating libraries with TensileCreateLibrary"
+        # To normalize special command line char handling between platforms.
+        VERBATIM
+        # To see progress vs buffering when built with ninja.
+        USES_TERMINAL)
 
       add_custom_target(${Tensile_VAR_PREFIX}_LIBRARY_TARGET
          DEPENDS "${Tensile_OUTPUT_PATH}/library"
          COMMAND ${CommandLine} "--verify-manifest"
-         COMMENT "Verifying files in ${Tensile_MANIFEST_FILE_PATH} were generated")
+         COMMENT "Verifying files in ${Tensile_MANIFEST_FILE_PATH} were generated"
+         VERBATIM)
   endif()
 
   if(Tensile_EMBED_LIBRARY)
@@ -279,7 +292,8 @@ function(TensileCreateLibraryFiles
           COMMAND ${CMAKE_COMMAND} -E copy
                   ${Tensile_EMBED_LIBRARY_SOURCE}
                   "${Tensile_OUTPUT_PATH}/library"
-          DEPENDS ${Tensile_EMBED_LIBRARY_SOURCE})
+          DEPENDS ${Tensile_EMBED_LIBRARY_SOURCE}
+          VERBATIM)
   endif()
 
 endfunction()
-- 
2.47.1.windows.2

