From 8a2648c0df634317dd2ea3266e87a28aab1d9c12 Mon Sep 17 00:00:00 2001
From: Mika Laitio <mika.laitio@amd.com>
Date: Tue, 8 Jul 2025 12:27:14 +0000
Subject: [PATCH 1/2] add amd gpus

Signed-off-by: Mika Laitio <mika.laitio@amd.com>
---
 aiter/utility/dtypes.py | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/aiter/utility/dtypes.py b/aiter/utility/dtypes.py
index 5491e16..7348afc 100644
--- a/aiter/utility/dtypes.py
+++ b/aiter/utility/dtypes.py
@@ -5,8 +5,17 @@ from ..jit.utils.chip_info import get_gfx
 import argparse
 
 defaultDtypes = {
+    "gfx90a": {"fp8": torch.float8_e4m3fn},
     "gfx942": {"fp8": torch.float8_e4m3fnuz},
     "gfx950": {"fp8": torch.float8_e4m3fn},
+    "gfx1100": {"fp8": torch.float8_e4m3fn},
+    "gfx1101": {"fp8": torch.float8_e4m3fn},
+    "gfx1102": {"fp8": torch.float8_e4m3fn},
+    "gfx1103": {"fp8": torch.float8_e4m3fn},
+    "gfx1150": {"fp8": torch.float8_e4m3fn},
+    "gfx1151": {"fp8": torch.float8_e4m3fn},
+    "gfx1200": {"fp8": torch.float8_e4m3fn},
+    "gfx1201": {"fp8": torch.float8_e4m3fn},
 }
 
 
-- 
2.43.0

