From ea73cb3e1ee1e156b1db3c72d6b16338f33c3ef2 Mon Sep 17 00:00:00 2001
From: Scott <scott.todd0@gmail.com>
Date: Tue, 8 Apr 2025 16:10:28 -0700
Subject: [PATCH 5/7] Fix CMAKE_INSTALL_PREFIX init behavior on Win32.

---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 49f03c8c..1d224c3e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -334,7 +334,9 @@ set( CPACK_RESOURCE_FILE_LICENSE "${CMAKE_CURRENT_SOURCE_DIR}/LICENSE.md" )
 set( CPACK_RPM_PACKAGE_LICENSE "MIT and BSD")
 
 if (WIN32)
-  SET( CMAKE_INSTALL_PREFIX "C:/hipSDK" CACHE PATH "Install path" FORCE )
+  if( CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT )
+    SET( CMAKE_INSTALL_PREFIX "C:/hipSDK" CACHE PATH "Install path" FORCE )
+  endif()
   SET( INSTALL_PREFIX "C:/hipSDK" )
   SET( CPACK_SET_DESTDIR FALSE )
   SET( CPACK_PACKAGE_INSTALL_DIRECTORY "C:/hipSDK" )
-- 
2.43.0

