From e6ff164f077169d608adce19ef25c152f0547ae5 Mon Sep 17 00:00:00 2001
From: Scott <scott.todd0@gmail.com>
Date: Tue, 29 Apr 2025 13:25:16 -0700
Subject: [PATCH 5/5] Fix include ordering for Windows compatibility.

---
 clients/gtest/rocblas_test.cpp | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/clients/gtest/rocblas_test.cpp b/clients/gtest/rocblas_test.cpp
index f9f13ae1..46a9c4d3 100644
--- a/clients/gtest/rocblas_test.cpp
+++ b/clients/gtest/rocblas_test.cpp
@@ -21,12 +21,11 @@
  * ************************************************************************ */
 
 #include "rocblas_test.hpp"
-#include "client_utility.hpp"
 
-#include <cstdlib>
-#include <exception>
-#include <regex>
 #ifdef WIN32
+// Must include windows.h before dependent headers.
+// Specifically this must be before client_utility.hpp uses `#pragma GCC poison`
+// to poison 'ctime' and 'abort', as those are used in standard library headers.
 #include <windows.h>
 #define strcasecmp(A, B) _stricmp(A, B)
 #else
@@ -34,6 +33,12 @@
 #include <unistd.h>
 #endif
 
+#include "client_utility.hpp"
+
+#include <cstdlib>
+#include <exception>
+#include <regex>
+
 /*********************************************
  * thread pool functions
  *********************************************/
-- 
2.43.0

