From 6f2b70121a898c38924936e1ae3109fb58980816 Mon Sep 17 00:00:00 2001
From: geomin12 <geomin12@amd.com>
Date: Thu, 24 Jul 2025 16:58:26 -0700
Subject: [PATCH] Removing gtest download

---
 clients/cmake/build-gtest.cmake | 22 +---------------------
 1 file changed, 1 insertion(+), 21 deletions(-)

diff --git a/clients/cmake/build-gtest.cmake b/clients/cmake/build-gtest.cmake
index a5a2b83e..bac800ac 100644
--- a/clients/cmake/build-gtest.cmake
+++ b/clients/cmake/build-gtest.cmake
@@ -23,25 +23,5 @@ include( ExternalProject )
 option( BUILD_GTEST "Download and build GoogleTest" OFF )
 
 if( NOT BUILD_GTEST )
-    find_package( GTest 1.16.0 )
-endif()
-
-if( (BUILD_GTEST OR NOT GTEST_FOUND) AND (NOT TARGET gtest) )
-  set(GTEST_INCLUDE_DIRS
-      ${CMAKE_CURRENT_BINARY_DIR}/src/gtest/googletest/include)
-  set(GTEST_LIBRARIES
-      ${CMAKE_CURRENT_BINARY_DIR}/src/gtest-build/lib/${CMAKE_STATIC_LIBRARY_PREFIX}gtest${CMAKE_STATIC_LIBRARY_SUFFIX}
-      ${CMAKE_CURRENT_BINARY_DIR}/src/gtest-build/lib/${CMAKE_STATIC_LIBRARY_PREFIX}gtest_main${CMAKE_STATIC_LIBRARY_SUFFIX})
-  
-  set(GTEST_SRC_URL https://github.com/google/googletest/archive/release-1.11.0.tar.gz CACHE STRING "Location of GTest source code")
-  set(GTEST_SRC_SHA256 b4870bf121ff7795ba20d20bcdd8627b8e088f2d1dab299a031c1034eddc93d5 CACHE STRING "SHA256 hash of GTest source code")
-  
-  ExternalProject_Add(gtest
-                      URL ${GTEST_SRC_URL}
-                      URL_HASH SHA256=${GTEST_SRC_SHA256}
-                      PREFIX ${CMAKE_CURRENT_BINARY_DIR}
-                      CMAKE_ARGS -DCMAKE_CXX_COMPILER_LAUNCHER=${CMAKE_CXX_COMPILER_LAUNCHER}
-                      INSTALL_COMMAND ""
-                      BUILD_BYPRODUCTS ${GTEST_LIBRARIES})
-  ExternalProject_Get_Property( gtest source_dir binary_dir )  
+    find_package( GTest )
 endif()
-- 
2.47.1.windows.2

