From 490ef7e19be9df99f7e23fce4b89e8d17e75464a Mon Sep 17 00:00:00 2001
From: Scott <scott.todd0@gmail.com>
Date: Wed, 2 Apr 2025 14:57:58 -0700
Subject: [PATCH] Fix CMAKE_INSTALL_PREFIX init behavior on Win32.

---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9559e515..391a5653 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -288,7 +288,9 @@ endif( )
 if(WIN32)
   set(CPACK_SOURCE_GENERATOR "ZIP")
   set(CPACK_GENERATOR "ZIP")
-  set(CMAKE_INSTALL_PREFIX "C:/hipSDK" CACHE PATH "Install path" FORCE)
+  if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+    set(CMAKE_INSTALL_PREFIX "C:/hipSDK" CACHE PATH "Install path" FORCE)
+  endif()
   set(INSTALL_PREFIX "C:/hipSDK")
   set(CPACK_SET_DESTDIR OFF)
   set(CPACK_PACKAGE_INSTALL_DIRECTORY "C:/hipSDK")
-- 
2.47.1.windows.2

