From d946ec237705a9a18d56a4516e4170da7bf7f680 Mon Sep 17 00:00:00 2001
From: Scott <scott.todd0@gmail.com>
Date: Wed, 2 Apr 2025 14:42:25 -0700
Subject: [PATCH 2/2] Fix CMAKE_INSTALL_PREFIX init behavior on Win32.

---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1699c029..c65552b8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -326,7 +326,9 @@ set(CPACK_RPM_PACKAGE_LICENSE "MIT")
 if(WIN32)
   set(CPACK_SOURCE_GENERATOR "ZIP")
   set(CPACK_GENERATOR "ZIP")
-  set(CMAKE_INSTALL_PREFIX "C:/hipSDK" CACHE PATH "Install path" FORCE)
+  if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+    set(CMAKE_INSTALL_PREFIX "C:/hipSDK" CACHE PATH "Install path" FORCE)
+  endif()
   set(INSTALL_PREFIX "C:/hipSDK")
   set(CPACK_SET_DESTDIR OFF)
   set(CPACK_PACKAGE_INSTALL_DIRECTORY "C:/hipSDK")
-- 
2.47.1.windows.2

