From 138c8d3ce1eae26d680f863b919f5eda3af4efb8 Mon Sep 17 00:00:00 2001
From: Scott <scott.todd0@gmail.com>
Date: Thu, 1 May 2025 11:29:55 -0700
Subject: [PATCH 16/17] Fix OpenBLAS package name for Windows.

---
 projects/hipblas/clients/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/projects/hipblas/clients/CMakeLists.txt b/projects/hipblas/clients/CMakeLists.txt
index 41364efb12..9f7c0cf0d3 100644
--- a/projects/hipblas/clients/CMakeLists.txt
+++ b/projects/hipblas/clients/CMakeLists.txt
@@ -204,7 +204,7 @@ if( BUILD_CLIENTS_BENCHMARKS OR BUILD_CLIENTS_TESTS)
                     NO_DEFAULT_PATH
                   )
       if (NOT BLAS_LIBRARY)
-        find_package( OPENBLAS CONFIG REQUIRED )
+        find_package( OpenBLAS CONFIG REQUIRED )
         set( BLAS_LIBRARY OpenBLAS::OpenBLAS )
         set( BLAS_INCLUDE_DIR "" )
       endif()
-- 
2.43.0

