From a785e82c8100d36a3d1eeafa74a9fbc356f15954 Mon Sep 17 00:00:00 2001
From: Scott Todd <scott.todd0@gmail.com>
Date: Wed, 21 May 2025 15:27:09 -0700
Subject: [PATCH] Guard against missing directories in Tensile Validators.py.

---
 tensilelite/Tensile/Toolchain/Validators.py | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/tensilelite/Tensile/Toolchain/Validators.py b/tensilelite/Tensile/Toolchain/Validators.py
index 3911a2d1..fd5dab53 100644
--- a/tensilelite/Tensile/Toolchain/Validators.py
+++ b/tensilelite/Tensile/Toolchain/Validators.py
@@ -50,10 +50,14 @@ def _windowsLatestRocmBin(path: Union[Path, str]) -> Path:
     Returns:
         The path to the ROCm bin directory for the latest ROCm version.
         Typically of the form ``C:/Program Files/AMD/ROCm/X.Y/bin``.
+        May return ``None`` if no ``X.Y`` subdirectories exist, perhaps due to
+        a partial uninstall.
     """
     path = Path(path)
     pattern = re.compile(r"^\d+\.\d+$")
-    versions = filter(lambda d: d.is_dir() and pattern.match(d.name), path.iterdir())
+    versions = list(filter(lambda d: d.is_dir() and pattern.match(d.name), path.iterdir()))
+    if len(versions) == 0:
+        return None
     latest = max(versions, key=lambda d: tuple(map(int, d.name.split("."))))
     return latest / "bin"
 
@@ -67,7 +71,9 @@ def _windowsSearchPaths() -> List[Path]:
         searchPaths.extend(hipPaths)
 
     if Path(defaultPath).exists():
-        searchPaths.append(_windowsLatestRocmBin(defaultPath))
+        latestRocmBin = _windowsLatestRocmBin(defaultPath)
+        if latestRocmBin:
+            searchPaths.append(latestRocmBin)
 
     if os.environ.get("PATH"):
         envPath = [Path(p) for p in os.environ["PATH"].split(os.pathsep)]
-- 
2.45.1.windows.1

