From 294d4db2ad360dd6603cf7358d3410ad19f75e72 Mon Sep 17 00:00:00 2001
From: Marius Brehler <marius.brehler@amd.com>
Date: Tue, 18 Mar 2025 12:14:21 +0000
Subject: [PATCH 6/7] Build `hipblas-sequence` on non-Ubuntu systems

---
 clients/benchmarks/CMakeLists.txt | 64 +++++++++++++++----------------
 1 file changed, 30 insertions(+), 34 deletions(-)

diff --git a/clients/benchmarks/CMakeLists.txt b/clients/benchmarks/CMakeLists.txt
index 39aa4e8c..1e0f8940 100644
--- a/clients/benchmarks/CMakeLists.txt
+++ b/clients/benchmarks/CMakeLists.txt
@@ -118,42 +118,38 @@ add_executable( hipblaslt-bench-extop-matrixtransform client_extop_matrixtransfo
 add_executable( hipblaslt-bench-extop-softmax client_extop_softmax.cpp ../common/hipblaslt_random.cpp)
 add_executable( hipblaslt-bench-extop-amax client_extop_amax.cpp  ../common/hipblaslt_random.cpp)
 set(ext_bench_list_all hipblaslt-api-overhead hipblaslt-bench-groupedgemm-fixed-mk hipblaslt-bench-extop-layernorm hipblaslt-bench-extop-matrixtransform hipblaslt-bench-extop-softmax hipblaslt-bench-extop-amax)
-# Currently only build this for Ubuntu because other distro's llvm-dev version is not matching the rocm's version.
 if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
-  file(READ "/etc/os-release" OS_RELEASE)
-  if(OS_RELEASE MATCHES "Ubuntu")
-    add_executable( hipblaslt-sequence client_sequence.cpp ../common/hipblaslt_random.cpp)
-    find_package(LLVM 13.0 QUIET CONFIG)
-    if(NOT LLVM_FOUND)
-        find_package(LLVM 12.0 QUIET CONFIG)
-        if(NOT LLVM_FOUND)
-            find_package(LLVM REQUIRED CONFIG)
-        endif()
-    endif()
-    find_library(LLVMObjectYAML_LIBRARY
-      NAMES LLVMObjectYAML
-      PATHS ${LLVM_LIBRARY_DIR})
-    if(LLVMObjectYAML_LIBRARY)
-      target_link_libraries(hipblaslt-sequence PRIVATE LLVMObjectYAML )
-    else()
-      target_link_libraries(hipblaslt-sequence PRIVATE LLVM )
-    endif()
-    target_include_directories(hipblaslt-sequence PRIVATE ${LLVM_INCLUDE_DIRS})
-    set(ext_bench_list_all ${ext_bench_list_all} hipblaslt-sequence)
-    set( HIPBLASLT_SEQUENCE_YAML "${PROJECT_BINARY_DIR}/staging/sequence.yaml")
-    add_custom_command( OUTPUT "${HIPBLASLT_SEQUENCE_YAML}"
-                        COMMAND ${CMAKE_COMMAND} -E copy sequence.yaml "${HIPBLASLT_SEQUENCE_YAML}"
-                        DEPENDS sequence.yaml
-                        WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}" )
-
-    add_custom_target( hipblaslt-sequence-yaml DEPENDS "${HIPBLASLT_SEQUENCE_YAML}" )
-    add_dependencies( hipblaslt-sequence hipblaslt-sequence-yaml )
-    rocm_install(
-      FILES ${HIPBLASLT_SEQUENCE_YAML}
-      DESTINATION "${CMAKE_INSTALL_BINDIR}"
-      COMPONENT benchmarks
-    )
+  add_executable( hipblaslt-sequence client_sequence.cpp ../common/hipblaslt_random.cpp)
+  find_package(LLVM 13.0 QUIET CONFIG)
+  if(NOT LLVM_FOUND)
+      find_package(LLVM 12.0 QUIET CONFIG)
+      if(NOT LLVM_FOUND)
+          find_package(LLVM REQUIRED CONFIG)
+      endif()
   endif()
+  find_library(LLVMObjectYAML_LIBRARY
+    NAMES LLVMObjectYAML
+    PATHS ${LLVM_LIBRARY_DIR})
+  if(LLVMObjectYAML_LIBRARY)
+    target_link_libraries(hipblaslt-sequence PRIVATE LLVMObjectYAML )
+  else()
+    target_link_libraries(hipblaslt-sequence PRIVATE LLVM )
+  endif()
+  target_include_directories(hipblaslt-sequence PRIVATE ${LLVM_INCLUDE_DIRS})
+  set(ext_bench_list_all ${ext_bench_list_all} hipblaslt-sequence)
+  set( HIPBLASLT_SEQUENCE_YAML "${PROJECT_BINARY_DIR}/staging/sequence.yaml")
+  add_custom_command( OUTPUT "${HIPBLASLT_SEQUENCE_YAML}"
+                      COMMAND ${CMAKE_COMMAND} -E copy sequence.yaml "${HIPBLASLT_SEQUENCE_YAML}"
+                      DEPENDS sequence.yaml
+                      WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}" )
+
+  add_custom_target( hipblaslt-sequence-yaml DEPENDS "${HIPBLASLT_SEQUENCE_YAML}" )
+  add_dependencies( hipblaslt-sequence hipblaslt-sequence-yaml )
+  rocm_install(
+    FILES ${HIPBLASLT_SEQUENCE_YAML}
+    DESTINATION "${CMAKE_INSTALL_BINDIR}"
+    COMPONENT benchmarks
+  )
 endif()
 
 foreach( exe ${ext_bench_list_all} )
-- 
2.43.0

