From baf924dd915b9e7959f13c88da0dbbcd1d903482 Mon Sep 17 00:00:00 2001
From: Scott Todd <scott.todd0@gmail.com>
Date: Wed, 16 Jul 2025 13:37:23 -0700
Subject: [PATCH] Allow rocblas_DIR env var to be unset on Windows.

---
 clients/gtest/CMakeLists.txt | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/clients/gtest/CMakeLists.txt b/clients/gtest/CMakeLists.txt
index eaff149..3d95a2c 100755
--- a/clients/gtest/CMakeLists.txt
+++ b/clients/gtest/CMakeLists.txt
@@ -197,11 +197,15 @@ if(WIN32)
       ARGS -E copy ${file_i} $<TARGET_FILE_DIR:rocsolver-test>
     )
   endforeach()
-  add_custom_command(TARGET rocsolver-test
-    POST_BUILD
-    COMMAND ${CMAKE_COMMAND}
-    ARGS -E copy_directory $ENV{rocblas_DIR}/bin/rocblas/library $<TARGET_FILE_DIR:rocsolver-test>/library
-  )
+  if(DEFINED ENV{rocblas_DIR})
+    add_custom_command(TARGET rocsolver-test
+      POST_BUILD
+      COMMAND ${CMAKE_COMMAND}
+      ARGS -E copy_directory $ENV{rocblas_DIR}/bin/rocblas/library $<TARGET_FILE_DIR:rocsolver-test>/library
+    )
+  else()
+    message(WARNING "rocblas_DIR not set. rocBLAS Tensile runtime kernels will not be copied to client staging directory (insitu testing affected).")
+  endif()
 endif()
 
 target_link_libraries(rocsolver-test PRIVATE
-- 
2.47.1.windows.2

