From 77a75ffce90524e6e5fafed0759d30b63a156286 Mon Sep 17 00:00:00 2001
From: Stella Laurenzo <stellaraccident@gmail.com>
Date: Tue, 1 Apr 2025 14:52:15 -0700
Subject: [PATCH 4/5] Pin cmake<4 since v4 broke old pytorch builds.

---
 requirements.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/requirements.txt b/requirements.txt
index 642d78e4f6..d02e1e9d7a 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -18,4 +18,4 @@ lintrunner
 ninja
 packaging
 optree>=0.13.0
-cmake
+cmake<4
-- 
2.43.0

