From e178e55e601d7595d1476571dba8020ee2cc492a Mon Sep 17 00:00:00 2001
From: Scott <scott.todd0@gmail.com>
Date: Thu, 1 May 2025 11:29:42 -0700
Subject: [PATCH 5/7] Setup Fortran on Windows.

---
 CMakeLists.txt | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 42dfd62..2322567 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,12 +42,11 @@ if( NOT DEFINED CMAKE_CONFIGURATION_TYPES AND NOT DEFINED CMAKE_BUILD_TYPE )
   set( CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel." )
 endif()
 
-if (NOT WIN32)
-    if ( NOT DEFINED CMAKE_Fortran_COMPILER AND NOT DEFINED ENV{FC} )
-      set( CMAKE_Fortran_COMPILER  "gfortran" )
-    endif()
-    set( fortran_language "Fortran" )
-endif( )
+if ( NOT DEFINED CMAKE_Fortran_COMPILER AND NOT DEFINED ENV{FC} )
+  set( CMAKE_Fortran_COMPILER  "gfortran" )
+endif()
+
+set( fortran_language "Fortran" )
 
 project( hipblas LANGUAGES CXX ${fortran_language} )
 
-- 
2.47.1.windows.2

