From 341468c4728fd6864facdd27d79cd20778e63aa4 Mon Sep 17 00:00:00 2001
From: Scott <scott.todd0@gmail.com>
Date: Wed, 30 Apr 2025 12:03:23 -0700
Subject: [PATCH 6/6] Replace ${python} with official ${Python3_EXECUTABLE}
 variable.

---
 clients/gtest/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/clients/gtest/CMakeLists.txt b/clients/gtest/CMakeLists.txt
index 9434c4d..82d97da 100644
--- a/clients/gtest/CMakeLists.txt
+++ b/clients/gtest/CMakeLists.txt
@@ -234,7 +234,7 @@ if( BUILD_WITH_SOLVER )
 endif()
 
 add_custom_command( OUTPUT "${HIPBLAS_TEST_DATA}"
-                    COMMAND ${python} ../common/hipblas_gentest.py -I ../include hipblas_gtest.yaml -o "${HIPBLAS_TEST_DATA}"
+                    COMMAND ${Python3_EXECUTABLE} ../common/hipblas_gentest.py -I ../include hipblas_gtest.yaml -o "${HIPBLAS_TEST_DATA}"
                     DEPENDS ../common/hipblas_gentest.py ../include/hipblas_common.yaml "${HIPBLAS_AUX_YAML_DATA}" "${HIPBLAS_L1_YAML_DATA}" "${HIPBLAS_L2_YAML_DATA}" "${HIPBLAS_L3_YAML_DATA}" "${HIPBLAS_EX_YAML_DATA}" "${HIPBLAS_SOLVER_YAML_DATA}" hipblas_gtest.yaml
                     WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}" )
 
-- 
2.45.1.windows.1

