From 259e93ed7697ea2ae2ba2e236c142c70255e88c1 Mon Sep 17 00:00:00 2001
From: Scott <scott.todd0@gmail.com>
Date: Thu, 1 May 2025 11:29:55 -0700
Subject: [PATCH 6/7] Fix OpenBLAS package name for Windows.

---
 clients/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/clients/CMakeLists.txt b/clients/CMakeLists.txt
index 86b470b..650dc34 100644
--- a/clients/CMakeLists.txt
+++ b/clients/CMakeLists.txt
@@ -184,7 +184,7 @@ if( BUILD_CLIENTS_BENCHMARKS OR BUILD_CLIENTS_TESTS)
                     NO_DEFAULT_PATH
                   )
       if (NOT BLAS_LIBRARY)
-        find_package( OPENBLAS CONFIG REQUIRED )
+        find_package( OpenBLAS CONFIG REQUIRED )
         set( BLAS_LIBRARY OpenBLAS::OpenBLAS )
         set( BLAS_INCLUDE_DIR "" )
       endif()
-- 
2.43.0

