From 1827dcd374be1448f6ced7b6502cca75c39ccdd1 Mon Sep 17 00:00:00 2001
From: Scott Todd <scott.todd0@gmail.com>
Date: Thu, 24 Jul 2025 13:02:18 -0700
Subject: [PATCH] Fix LoadHIP handling of environment variable paths on
 Windows.

---
 cmake/LoadHIP.cmake | 34 +++++++++++++++++-----------------
 1 file changed, 17 insertions(+), 17 deletions(-)

diff --git a/cmake/LoadHIP.cmake b/cmake/LoadHIP.cmake
index 9227e43d..3ca1befc 100644
--- a/cmake/LoadHIP.cmake
+++ b/cmake/LoadHIP.cmake
@@ -7,7 +7,7 @@ if(NOT DEFINED ENV{ROCM_PATH})
     set(ROCM_PATH C:/opt/rocm)
   endif()
 else()
-  set(ROCM_PATH $ENV{ROCM_PATH})
+  file(TO_CMAKE_PATH "$ENV{ROCM_PATH}" ROCM_PATH)
 endif()
 
 # HIP_PATH
@@ -18,7 +18,7 @@ if(NOT DEFINED ENV{HIP_PATH})
     set(HIP_PATH ${ROCM_PATH})
   endif()
 else()
-  set(HIP_PATH $ENV{HIP_PATH})
+  file(TO_CMAKE_PATH "$ENV{HIP_PATH}" HIP_PATH)
 endif()
 
 if(NOT EXISTS ${HIP_PATH})
@@ -29,47 +29,47 @@ endif()
 if(NOT DEFINED ENV{HCC_PATH})
   set(HCC_PATH ${ROCM_PATH}/hcc)
 else()
-  set(HCC_PATH $ENV{HCC_PATH})
+  file(TO_CMAKE_PATH "$ENV{HCC_PATH}" HCC_PATH)
 endif()
 
 # HSA_PATH
 if(NOT DEFINED ENV{HSA_PATH})
   set(HSA_PATH ${ROCM_PATH}/hsa)
 else()
-  set(HSA_PATH $ENV{HSA_PATH})
+  file(TO_CMAKE_PATH "$ENV{HSA_PATH}" HSA_PATH)
 endif()
 
 # ROCBLAS_PATH
 if(NOT DEFINED ENV{ROCBLAS_PATH})
   set(ROCBLAS_PATH ${ROCM_PATH}/rocblas)
 else()
-  set(ROCBLAS_PATH $ENV{ROCBLAS_PATH})
+  file(TO_CMAKE_PATH "$ENV{ROCBLAS_PATH}" ROCBLAS_PATH)
 endif()
 
 # ROCFFT_PATH
 if(NOT DEFINED ENV{ROCFFT_PATH})
   set(ROCFFT_PATH ${ROCM_PATH}/rocfft)
 else()
-  set(ROCFFT_PATH $ENV{ROCFFT_PATH})
+  file(TO_CMAKE_PATH "$ENV{ROCFFT_PATH}" ROCFFT_PATH)
 endif()
 
 # HIPFFT_PATH
 if(NOT DEFINED ENV{HIPFFT_PATH})
   set(HIPFFT_PATH ${ROCM_PATH}/hipfft)
 else()
-  set(HIPFFT_PATH $ENV{HIPFFT_PATH})
+  file(TO_CMAKE_PATH "$ENV{HIPFFT_PATH}" HIPFFT_PATH)
 endif()
 
 # HIPSPARSE_PATH
 if(NOT DEFINED ENV{HIPSPARSE_PATH})
   set(HIPSPARSE_PATH ${ROCM_PATH}/hipsparse)
 else()
-  set(HIPSPARSE_PATH $ENV{HIPSPARSE_PATH})
+  file(TO_CMAKE_PATH "$ENV{HIPSPARSE_PATH}" HIPSPARSE_PATH)
 endif()
 
 # THRUST_PATH
 if(DEFINED ENV{THRUST_PATH})
-  set(THRUST_PATH $ENV{THRUST_PATH})
+  file(TO_CMAKE_PATH "$ENV{THRUST_PATH}" THRUST_PATH)
 else()
   set(THRUST_PATH ${ROCM_PATH}/include)
 endif()
@@ -78,56 +78,56 @@ endif()
 if(NOT DEFINED ENV{HIPRAND_PATH})
   set(HIPRAND_PATH ${ROCM_PATH}/hiprand)
 else()
-  set(HIPRAND_PATH $ENV{HIPRAND_PATH})
+  file(TO_CMAKE_PATH "$ENV{HIPRAND_PATH}" HIPRAND_PATH)
 endif()
 
 # ROCRAND_PATH
 if(NOT DEFINED ENV{ROCRAND_PATH})
   set(ROCRAND_PATH ${ROCM_PATH}/rocrand)
 else()
-  set(ROCRAND_PATH $ENV{ROCRAND_PATH})
+  file(TO_CMAKE_PATH "$ENV{ROCRAND_PATH}" ROCRAND_PATH)
 endif()
 
 # MIOPEN_PATH
 if(NOT DEFINED ENV{MIOPEN_PATH})
   set(MIOPEN_PATH ${ROCM_PATH}/miopen)
 else()
-  set(MIOPEN_PATH $ENV{MIOPEN_PATH})
+  file(TO_CMAKE_PATH "$ENV{MIOPEN_PATH}" MIOPEN_PATH)
 endif()
 
 # RCCL_PATH
 if(NOT DEFINED ENV{RCCL_PATH})
   set(RCCL_PATH ${ROCM_PATH}/rccl)
 else()
-  set(RCCL_PATH $ENV{RCCL_PATH})
+  file(TO_CMAKE_PATH "$ENV{RCCL_PATH}" RCCL_PATH)
 endif()
 
 # ROCPRIM_PATH
 if(NOT DEFINED ENV{ROCPRIM_PATH})
   set(ROCPRIM_PATH ${ROCM_PATH}/rocprim)
 else()
-  set(ROCPRIM_PATH $ENV{ROCPRIM_PATH})
+  file(TO_CMAKE_PATH "$ENV{ROCPRIM_PATH}" ROCPRIM_PATH)
 endif()
 
 # HIPCUB_PATH
 if(NOT DEFINED ENV{HIPCUB_PATH})
   set(HIPCUB_PATH ${ROCM_PATH}/hipcub)
 else()
-  set(HIPCUB_PATH $ENV{HIPCUB_PATH})
+  file(TO_CMAKE_PATH "$ENV{HIPCUB_PATH}" HIPCUB_PATH)
 endif()
 
 # ROCTHRUST_PATH
 if(NOT DEFINED ENV{ROCTHRUST_PATH})
   set(ROCTHRUST_PATH ${ROCM_PATH}/rocthrust)
 else()
-  set(ROCTHRUST_PATH $ENV{ROCTHRUST_PATH})
+  file(TO_CMAKE_PATH "$ENV{ROCTHRUST_PATH}" ROCTHRUST_PATH)
 endif()
 
 # ROCTRACER_PATH
 if(NOT DEFINED ENV{ROCTRACER_PATH})
   set(ROCTRACER_PATH ${ROCM_PATH}/roctracer)
 else()
-  set(ROCTRACER_PATH $ENV{ROCTRACER_PATH})
+  file(TO_CMAKE_PATH "$ENV{ROCTRACER_PATH}" ROCTRACER_PATH)
 endif()
 
 if(NOT DEFINED ENV{PYTORCH_ROCM_ARCH})
-- 
2.47.1.windows.2

