From 4099a852acb5c21f6d6f6c17367eb3ca20ab13e2 Mon Sep 17 00:00:00 2001
From: Scott <scott.todd0@gmail.com>
Date: Wed, 12 Mar 2025 13:49:43 -0700
Subject: [PATCH 1/2] PYTHON3_EXE should be Python3_EXECUTABLE. Either way, it
 is unused.

---
 toolchain-windows.cmake | 2 --
 1 file changed, 2 deletions(-)

diff --git a/toolchain-windows.cmake b/toolchain-windows.cmake
index f94d3c9..7364186 100644
--- a/toolchain-windows.cmake
+++ b/toolchain-windows.cmake
@@ -55,5 +55,3 @@ set(CMAKE_STATIC_LIBRARY_SUFFIX ".a")
 set(CMAKE_STATIC_LIBRARY_PREFIX "static_")
 set(CMAKE_SHARED_LIBRARY_SUFFIX ".dll")
 set(CMAKE_SHARED_LIBRARY_PREFIX "")
-
-set(PYTHON3_EXE python)
-- 
2.47.1.windows.2

